src/core/co_tmr.c

```diff
@@ -538,22 +538,24 @@ static void COTmrRemove(CO_TMR *tmr, CO_TMR_TIME *tx)
 
             /* loop through used timers in list until timer is removed */
             tn = tmr->Use;
-            do {
-                /* remove next timer in list */
-                if (tn->Next == tx) {
-                    tn->Next = tx->Next;
-
-                    /* timer was within list */
-                    if (tx->Next != 0) {
-                        tn->Next->Delta += tx->Delta;
+            if (tn != 0) {
+                do {
+                    /* remove next timer in list */
+                    if (tn->Next == tx) {
+                        tn->Next = tx->Next;
+
+                        /* timer was within list */
+                        if (tx->Next != 0) {
```

Comment:
this repeate of the conditional at line 541 suggests this should be a while loop rather than a do loop